Fix web compilation

danielgomezriconeon opened the pull request:
Fix:
- https://github.com/isar-community/isar/issues/80

Related to:
- https://github.com/isar-community/isar-community/issues/37

This is a continuation of https://github.com/isar-community/isar/issues/80 but for the new repo of the project.

TODO:
- [x] Make it compile on web fine
- [x] Make it compile for native too by having this change

danielgomezriconeon committed:
Fix web compilation

alanlanglois commented:
Hi there! Any update on this PR ?

vicenterusso commented:
Can't merge, at this moment this PR is 100% broken

danielgomezriconeon commented:
I was checking and it looks like the failure of the CI is related to the v3 branch itself, I was trying to fix that in:
- https://github.com/isar-community/isar-community/pull/67

@vicenterusso can you take a look?

vicenterusso commented:
> I was checking and it looks like the failure of the CI is related to the v3 branch itself, I was trying to fix that in:
> 
> * [Fix ci: always fails when it validate the format (linter issues are still causing it to fail) #67](https://github.com/isar-community/isar-community/pull/67)
> 
> @vicenterusso can you take a look?

The format and lint must be fixed, of course, but it is not blocking the release pipeline. At this point, it's expected to fail, but not all other tests.

mahdiahbabi commented:
Thanks for the work on maintaining Isar v3 🙏
do you plan to support and fix web compilation in v3, or is web support intentionally disabled at the moment?
@vicenterusso

vicenterusso commented:
> Thanks for the work on maintaining Isar v3 🙏 do you plan to support and fix web compilation in v3, or is web support intentionally disabled at the moment? @vicenterusso

Web support only in version 4 (beta version very soon)

nandy33up commented:
It's really helpful

danielgomezriconeon commented:
@vicenterusso I run `dart format` over the project, what else do you think its missing?

vicenterusso commented:
> @vicenterusso I run `dart format` over the project, what else do you think its missing?

All tests are failing, basically

danielgomezriconeon commented:
> > @vicenterusso I run `dart format` over the project, what else do you think its missing?
> 
> All tests are failing, basically

Is that not related to this missing piece? https://github.com/isar-community/isar-community/pull/67

danielgomezriconeon commented:
@vicenterusso friendly ping :)

vicenterusso commented:
> > > @vicenterusso I run `dart format` over the project, what else do you think its missing?
> > 
> > 
> > All tests are failing, basically
> 
> Is that not related to this missing piece? #67

Please create a branch containing only the essential fixes, leaving out any linting changes. I want to see the minimal file changes needed for all tests to pass

danielgomezriconeon commented:
@vicenterusso like that? I reverted the formatting code

danielgomezriconeon committed:
Add tests

danielgomezriconeon commented:
@vicenterusso friendly reminder

danielgomezriconeon committed:
Be clear about the intent of visibleForTesting

danielgomezriconeon committed:
Remove failing @visibleForTesting

danielgomezriconeon committed:
Fix format

danielgomezriconeon committed:
fix(generator): revert to raw xxh3 IDs for native parity

The generateWebSafeId wrapper in the generator produced IDs that
didn't match the native Rust core's raw xxh3_64 computation, causing
1181 test failures ("Collection id is invalid").

- Revert ObjectInfo.id, ObjectIndex.id, ObjectLink.id to raw xxh3
- Move generateWebSafeId to web runtime for future web support
- Move web_safe_id_test to isar_community package
